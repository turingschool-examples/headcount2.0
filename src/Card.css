
.card {
  width: 150px;
  text-align: center;
  margin: 2%;
  background: white;
  opacity: .86;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 3px 3px 3px #7c7c7c;
}

h1 {
  font-size: 20px;
}

.data {
  font-size: 15px;
  text-align: left;
  font-weight: bold;
}

.less-than {
  color: #ea6512;
}

.greater-than {
  color: #4f4f4f;
}
