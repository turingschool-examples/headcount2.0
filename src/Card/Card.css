.card {
    box-shadow: 5px 5px 8px;
    width: 185px;
    height: 233px;
    border: 1px #000000 solid;
    margin: 7px;
}

ul {
  list-style-type: none;
}

li {
  font-size: 12px;
}

h3 {
  text-align: center;
  font-size: 15px;
  padding: 0 7px;
}