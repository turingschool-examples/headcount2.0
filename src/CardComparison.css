.comparison-wrapper {
  border: thin solid black;
  width: 65%;
  margin: 0px auto 10px;
  
}

.card-comparison-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  font-family: 'Raleway', sans-serif;
}

.comparison-heading {
  font-size: 1.5em;
  text-align: center;
  margin: 20px;
  font-family: 'Raleway', sans-serif;

}

.comparison-data {
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 10px;
}

.district-name {
  font-size: 1em;
  margin: 10;

}

.district-avg {
  margin: 0px;
  padding-left: 15px;
}

.comparison-value {
  font-weight: 400;
  margin: 10;
}

.clear-btn {
  width: 75px;
  height: 28px;
  border: 1px solid #0095A9;
  color: #0095A9;
  border-radius: 5px;
  cursor: pointer;
  font-size: 14px;
  font-weight: 600;
}