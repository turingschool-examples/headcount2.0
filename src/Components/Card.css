.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}