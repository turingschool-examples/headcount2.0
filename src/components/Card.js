import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
