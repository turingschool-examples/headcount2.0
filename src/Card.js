import React from 'react';
import './Card.css';

const Card = ({schoolNames, helper}) => {
  const schoolName = helper.data[schoolNames].location;
  const annualData = helper.data[schoolNames].annualData;
  console.log(annualData);
  return (
    <div className="card">
      <h1>{schoolName}</h1>
      {Object.keys(annualData).map((year, index) => {
        if (annualData[year] <= 0.5) {
          return (
            <ul>
              <li><p className="less-than data">{year}: {annualData[year]}</p></li>
            </ul>
          )
        } else {
          return (
            <ul>
              <li><p className="greater-than data">{year}: {annualData[year]}</p></li>
            </ul>
          )
        }
      })}
    </div>
  )
}

export default Card;
