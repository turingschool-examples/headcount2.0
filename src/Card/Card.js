import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: this.props.clickedState || false
    };
  }

  handleCardClick = () => {
    
    if (this.props.comparedCards.length === 2  && 
      !this.props.comparedCards.find(district => {
        return district.district.location === this.props.district.location;
      })) {
      return;
    }
    this.setState({ clicked: !this.state.clicked });
    this.props.updateCompareState(this.state.clicked, this.props);
  }

  render() {
    let districtKeys = Object.keys(this.props.district.stats);
    const listItems = districtKeys.map(( stat, index ) => {
      return (
        <li 
          key={ index }
          className={ this.props.district.stats[stat] > 0.5 ? 'greater-than' : 'less-than'}
        > { `${stat}: ${this.props.district.stats[stat]}` } </li>
      );
    });
  
    return (
      <div 
        className={this.state.clicked ? 'card clicked' : 'card'} 
        onClick={ this.handleCardClick }
      > 
        <h3>
          { this.props.district.location }
        </h3>
        <ul>
          { listItems }
        </ul>
      </div>
    );
  }
}

Card.propTypes = {
  district: PropTypes.string.isRequired,
  comparedCards: PropTypes.arrayOf(PropTypes.object),
  updateCompareState: PropTypes.func.isRequired,
  clickedState: PropTypes.bool
};



export default Card;