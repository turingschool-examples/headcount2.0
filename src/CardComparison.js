import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';

const borderColors = ['blue-border', 'green-border'];
const CardComparison = ( { comparisonArray, onCardClick } ) => (
  <div>
    {comparisonArray.map( (selectedDistrict, index) => {
      return (
        <Card
          // key={`selectedIndex:${index}`}
          key={selectedDistrict.location}
          borderColorClass={borderColors[index]}
          districtName={selectedDistrict.location}
          districtObject={selectedDistrict.data}
          onCardClick={onCardClick} />
      );
    })}
  </div>
);

export default CardComparison;

CardComparison.propTypes = {
  comparisonArray: PropTypes.array,
  onCardClick: PropTypes.func
};
