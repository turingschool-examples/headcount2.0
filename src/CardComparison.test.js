import React from 'react';
import { shallow } from 'enzyme';
import CardComparison from './CardComparison';
import Card from './Card';

describe('CardComparison', () => {
  let wrapper;
  let mockCompareData;
  let mockAnalysis;
  let compareDistrictDataMock;
  let clearComparisonMock;

  beforeEach(() => {
    mockCompareData = [{location: 'denver', stats: [{2001: 2}], selected: true},
      {location: 'boulder', stats: [{2002: 1}], selected: true}];
    mockAnalysis = {'AGATE 300': 0.909, 'ARICKAREE R-2': 0.739, compared: 1.23};
    compareDistrictDataMock = jest.fn();
    clearComparisonMock = jest.fn();
    wrapper = shallow(<CardComparison 
      compareData={mockCompareData}
      analysis={mockAnalysis}
      compareDistrictData={compareDistrictDataMock} 
      clearComparison={clearComparisonMock} />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all the cards', () => {
    expect(wrapper.find(Card).length).toEqual(2);
  });

  it('should call clearComparison when clear-btn is clicked', () => {
    wrapper.find('.clear-btn').simulate('click');
    expect(clearComparisonMock).toHaveBeenCalled();
  });
});

