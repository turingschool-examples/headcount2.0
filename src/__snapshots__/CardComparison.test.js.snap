// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComparison should match the snapshot with all data passed in correctly 1`] = `
<div
  className="comparison-wrapper"
>
  <h1
    className="comparison-heading"
  >
    School District Comparison
  </h1>
  <div
    className="card-comparison-container"
  >
    <Card
      clearComparison={[Function]}
      compareData={
        Array [
          Object {
            "location": "denver",
            "selected": true,
            "stats": Array [
              Object {
                "2001": 2,
              },
            ],
          },
          Object {
            "location": "boulder",
            "selected": true,
            "stats": Array [
              Object {
                "2002": 1,
              },
            ],
          },
        ]
      }
      district={
        Object {
          "location": "denver",
          "selected": true,
          "stats": Array [
            Object {
              "2001": 2,
            },
          ],
        }
      }
    />
    <Card
      clearComparison={[Function]}
      compareData={
        Array [
          Object {
            "location": "denver",
            "selected": true,
            "stats": Array [
              Object {
                "2001": 2,
              },
            ],
          },
          Object {
            "location": "boulder",
            "selected": true,
            "stats": Array [
              Object {
                "2002": 1,
              },
            ],
          },
        ]
      }
      district={
        Object {
          "location": "boulder",
          "selected": true,
          "stats": Array [
            Object {
              "2002": 1,
            },
          ],
        }
      }
    />
    <div
      className="comparison-data"
    >
      <h3
        className="district-name"
      >
        AGATE 300
      </h3>
      <p
        className="district-avg"
      >
        Average: 
        0.909
      </p>
      <h3
        className="district-name"
      >
        ARICKAREE R-2
      </h3>
      <p
        className="district-avg"
      >
        Average: 
        0.739
      </p>
      <h4
        className="comparison-value"
      >
        Compared Value: 
        1.23
      </h4>
      <button
        className="clear-btn"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </div>
</div>
`;
